x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts

```diff
@@ -47,38 +33,91 @@ export async function getUpgradeAssistantStatus(
 }
 
 // Reformats the index deprecations to an array of deprecation warnings extended with an index field.
-const getCombinedIndexInfos = (deprecations: DeprecationAPIResponse) =>
-  Object.keys(deprecations.index_settings).reduce((indexDeprecations, indexName) => {
-    return indexDeprecations.concat(
-      deprecations.index_settings[indexName].map(
-        (d) =>
-          ({
-            ...d,
-            index: indexName,
-            reindex: /Index created before/.test(d.message),
-            deprecatedIndexSettings: getIndexSettingDeprecations(d.message),
-          } as EnrichedDeprecationInfo)
-      )
-    );
-  }, [] as EnrichedDeprecationInfo[]);
-
-const getClusterDeprecations = (deprecations: DeprecationAPIResponse, isCloudEnabled: boolean) => {
-  const combined = deprecations.cluster_settings
+const getCombinedIndexInfos = async (
+  deprecations: DeprecationAPIResponse,
+  dataClient: IScopedClusterClient
+) => {
+  const indices = Object.keys(deprecations.index_settings).reduce(
+    (indexDeprecations, indexName) => {
+      return indexDeprecations.concat(
+        deprecations.index_settings[indexName].map(
+          (d) =>
+            ({
+              ...d,
+              index: indexName,
+              correctiveAction: getCorrectiveAction(d.message),
+            } as EnrichedDeprecationInfo)
+        )
+      );
+    },
+    [] as EnrichedDeprecationInfo[]
+  );
+
+  const indexNames = indices.map(({ index }) => index!);
+
+  // If we have found deprecation information for index/indices
+  // check whether the index is open or closed.
+  if (indexNames.length) {
+    const indexStates = await esIndicesStateCheck(dataClient.asCurrentUser, indexNames);
+
+    indices.forEach((indexData) => {
+      if (indexData.correctiveAction?.type === 'reindex') {
+        indexData.correctiveAction.blockerForReindexing =
+          indexStates[indexData.index!] === 'closed' ? 'index-closed' : undefined;
+      }
+    });
+  }
+  return indices as EnrichedDeprecationInfo[];
+};
+
+const getClusterDeprecations = (deprecations: DeprecationAPIResponse) => {
+  const combinedDeprecations = deprecations.cluster_settings
     .concat(deprecations.ml_settings)
     .concat(deprecations.node_settings);
 
-  if (isCloudEnabled) {
```

Comment:
I confirmed with the cloud team that this is no longer applicable for the 8.0 upgrade; any cloud-specific logic in UA will be added separately once we know more.